tests/benchmarks/cuda/benchmark_propagator_cuda.cpp

```diff
@@ -0,0 +1,141 @@
+    // Set origin position of tracks
+    const point3 ori{0., 0., 0.};
+
+    // Loops of theta values ]0,pi[
+    for (unsigned int itheta = 0; itheta < theta_steps; ++itheta) {
+        scalar theta = 0.001 + itheta * (M_PI - 0.001) / theta_steps;
+        scalar sin_theta = std::sin(theta);
+        scalar cos_theta = std::cos(theta);
+
+        // Loops of phi values [-pi, pi]
+        for (unsigned int iphi = 0; iphi < phi_steps; ++iphi) {
+            // The direction
+            scalar phi = -M_PI + iphi * (2 * M_PI) / phi_steps;
+            scalar sin_phi = std::sin(phi);
+            scalar cos_phi = std::cos(phi);
+            vector3 mom{cos_phi * sin_theta, sin_phi * sin_theta, cos_theta};
+            mom = 10. * mom;
+
+            // intialize a track
+            free_track_parameters traj(ori, 0, mom, -1);
+
+            tracks.push_back(traj);
+        }
+    }
+}
+
+static void BM_PROPAGATOR_CPU(benchmark::State &state) {
+
+    // Create the toy geometry
+    detector_host_type det =
+        create_toy_geometry<darray, thrust::tuple, vecmem::vector,
+                            vecmem::jagged_vector>(host_mr, n_brl_layers,
+                                                   n_edc_layers);
+
+    // Set the magnetic field
+    const vector3 B{0, 0, 2 * unit_constants::T};
+    field_type B_field(B);
+
+    // Create RK stepper
+    rk_stepper_type s(B_field);
+
+    // Create navigator
+    navigator_host_type n(det);
+
+    // Create propagator
+    propagator_host_type p(std::move(s), std::move(n));
+
+    for (auto _ : state) {
+
+        state.PauseTiming();
+
+        // Get tracks
+        vecmem::vector<free_track_parameters> tracks(&host_mr);
+        fill_tracks(tracks, state.range(0), state.range(0));
+
+        state.ResumeTiming();
+
+        for (auto &track : tracks) {
+
+            void_propagator_inspector vi;
+
+            // Create the propagator state
+            propagator_host_type::state p_state(track);
+
+            // Run propagation
+            p.propagate(p_state, vi);
+        }
+    }
+}
+
+static void BM_PROPAGATOR_CUDA(benchmark::State &state) {
+
+    // Create the toy geometry
+    detector_host_type det =
+        create_toy_geometry<darray, thrust::tuple, vecmem::vector,
+                            vecmem::jagged_vector>(bp_mng_mr, n_brl_layers,
+                                                   n_edc_layers);
+
+    // Get detector data
+    auto det_data = get_data(det);
+
+    // vecmem copy helper object
+    vecmem::cuda::copy copy;
+
+    for (auto _ : state) {
+
+        state.PauseTiming();
+
+        // Get tracks
+        vecmem::vector<free_track_parameters> tracks(&bp_mng_mr);
+        fill_tracks(tracks, state.range(0), state.range(0));
+
+        state.ResumeTiming();
+
+        // Get tracks data
+        auto tracks_data = vecmem::get_data(tracks);
+
+        // Create navigator candidates buffer
+        auto candidates_buffer =
```

Comment:
Do we really want to measure this?

Reply:
Hmm, Maybe yes maybe no. If we fix the number of tracks over the run-time, we can utilize the same buffer. In that case we don't have to measure this. But let's leave it as it is until we make a decision on KF implementation